packages/x-components/src/x-modules/search/types.ts

```diff
@@ -0,0 +1,11 @@
+import { SearchRequest } from '@empathyco/x-adapter';
+
+/**
+ * A pageable search request.
+ *
+ * @public
+ */
+export interface PageableSearchRequest extends SearchRequest {
```

Comment:
In the declaration of `SearchRequest` it extends from `PageableRequest`, so creating the type `PageableSearchRequest` to make it actually pageable is kind of weird. I would say we can change the type `PageableRequest` to include the `page` field directly and get rid of this new type.

Reply:
It's for me too. When I saw that `SearchRequest` extends `PageableRequest` was really weird.

Reply:
I've renamed it to `InternalSearchRequest`, as this type is just some sort of "partial/incomplete" (not a `Typescript` `partial`) `SearchRequest` containing a `page` property, which is just used to ease the calculations of the `SearchRequest` properties `start` and `rows`.